src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+
+
+/**
+ * Represents an Appointment attached to a Person in AddressBook
+ * Guarantees: immutable
+ */
+public class Appointment implements Comparable<Appointment> {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Appointment constraints";
+
+    private static int idTracker = 1;
+    public final int appointmentId;
+    public final LocalDateTime appointmentDate;
+    public final int studentId;
+    public final String appointmentDescription;
+
+    /**
+     * Constructs a {@code Appointment}.
+     *
+     * @param appointmentId          unique id of the appointment.
+     * @param appointmentDate        date and time of the appointment.
+     * @param studentId              unique id of the student.
+     * @param appointmentDescription description of the appointment.
+     */
+    public Appointment(int appointmentId, LocalDateTime appointmentDate, int studentId, String appointmentDescription) {
+        this.appointmentId = appointmentId;
+        this.appointmentDate = appointmentDate;
+        // Student ID is the same as the person ID
+        this.studentId = studentId;
+        idTracker = appointmentId + 1;
+
+        this.appointmentDescription = appointmentDescription;
+        requireAllNonNull(appointmentDate, appointmentDescription);
+    }
```

Comment:
```suggestion
    public Appointment(int appointmentId, LocalDateTime appointmentDate, int studentId, String appointmentDescription) {
        requireAllNonNull(appointmentDate, appointmentDescription);
        this.appointmentId = appointmentId;
        this.appointmentDate = appointmentDate;
        // Student ID is the same as the person ID
        this.studentId = studentId;
        idTracker = appointmentId + 1;

        this.appointmentDescription = appointmentDescription;
    }
```

Reply:
This change simply moves requireAllNonNull to the first line of the constructor. This is to ensure code consistency since Person also does the same